### PR TITLE
Add `filename_space_subst` config option

### PR DESCRIPTION
**Pull request by skovati**
## Proposed change

Created a new PR since I deleted my fork and Github wouldn't let me reopen the old PR with a new fork.

Based on [previous feedback](https://github.com/renerocksai/telekasten.nvim/pull/195#issuecomment-1369783361), this PR adds a new configuration option, `filename_space_subst`, that replaces spaces in the filename when the title is used within the filename generation. This is done to avoid spaces in filenames, which often need to be escaped on unix-like systems. If set to `nil`, no substitution will be made. The option defaults to `nil` to avoid any changes for existing users.

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Documentation update

## Additional information

- This PR fixes or closes issue: N/A
- This PR is related to issue: N/A

## Checklist

- [x] I am running the **latest** version of the plugin.
- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] The code has been formatted using Stylua (a `.stylua.toml` file is provided)
- [x] The code has been checked with luacheck (a `.luacheckrc` file is provided)
- [x] The `README.md` has been updated according to this change.
- [x] The `doc/telekasten.txt` helpfile has been updated according to this change.

### CODE REVIEW

File: lua/telekasten.lua [hunk 1]
```diff
@@ -224,6 +227,10 @@ local function get_uuid(opts)
 end
 
 local function generate_note_filename(uuid, title)
+    if M.Cfg.filename_space_subst ~= nil then
+        title = title:gsub(" ", M.Cfg.filename_space_subst)
```
Comment (lambtho12) [inline lua/telekasten.lua:231]: I believe we should escape properly the title first using `escape` function. 
This function ensures everything works properly when we use `gsub`.

I do not have time now to dive deeper and really see why or how, but issues like #179 were simply solved by using this proper escape function first.

**Review by lambtho12 — CHANGES_REQUESTED** [review]
It is probably best to escape the title before using `gsub` to ensure special characters will not screw everything up.

File: lua/telekasten.lua (hunk 1 above)
Comment (skovati) [reply lua/telekasten.lua]: Change has been made! Just wrapped with `escape(title)` before `gsub`ing.

**Comment by skovati** [comment]
FYI, I discovered a small bug in my previous template refactor which I fixed in the latest commit https://github.com/renerocksai/telekasten.nvim/pull/198/commits/d45990956190423b8e343165d520777a0c1c0f08 of this PR

**Review by lambtho12 — APPROVED** [review]

**Comment by lambtho12** [comment]
Awesome! Thank you very much. Sorry for the delay

**Pull request merged**